spec/models/user_spec.rb

```diff
@@ -150,4 +150,53 @@
       expect(result).to eq [group_only_ally_belongs_to]
     end
   end
+
+  describe '#update_access_token' do
+    let!(:user) do
+      User.create(name: 'some name',
+                  email: 'some@user.com',
+                  password: 'asdfasdf',
+                  token: 'some token')
+    end
+
+    request = {
+      'refresh_token'   =>  nil,
+      'client_id'       =>  ENV['GOOGLE_CLIENT_ID'],
+      'client_secret'   =>  ENV['GOOGLE_CLIENT_SECRET'],
+      'grant_type'      =>  'refresh_token'
+    }
+
+    context 'when request is successful' do
+      before do
+        response = {
+          'access_token': 'MTQ0NjJkZmQ5OTM2NDE1ZTZjNGZmZjI3',
+          'token_type': 'bearer',
+          'expires_in': 3600,
+          'refresh_token': 'IwOGYzYTlmM2YxOTQ5MGE3YmNmMDFkNTVk',
+          'scope': 'create'
+        }.to_json
+
+        Net::HTTP.stub(:post_form).with(URI.parse(User::OAUTH_TOKEN_URL), request) { double(body: response) }
+      end
+
+      it 'returns a new access token' do
+        expect(user.update_access_token).to eq('MTQ0NjJkZmQ5OTM2NDE1ZTZjNGZmZjI3')
+      end
+    end
+
+    context 'when request is unsuccessful' do
+      before do
+        response = {
+          'error': 'inavalid request',
```

Comment:
Spellcheck: replace with invalid

Reply:
This is the response we get from google oauth. I think we might want to be consistent with their error response

Reply:
LOL really, wow. That makes sense to be consistent, but they should really fix that. You could probably file a PR if that code is open source.

Reply:
lol, I now get what you meant. Damn it!!  Okay `invalid` it is 🤣